x-pack/plugins/saved_objects_tagging/public/management/tag_management_page.tsx

```diff
@@ -160,13 +178,51 @@ export const TagManagementPage: FC<TagManagementPageParams> = ({
     [overlays, notifications, fetchTags, tagClient]
   );
 
+  const executeBulkAction = useCallback(
+    async (action: TagBulkAction) => {
+      try {
+        await action.execute(selectedTags.map(({ id }) => id));
+        if (action.refreshAfterExecute) {
+          await fetchTags();
+        }
+      } catch (e) {
+        notifications.toasts.addError(e, {
+          title: i18n.translate('xpack.savedObjectsTagging.notifications.bulkActionError', {
+            defaultMessage: 'An error occurred',
+          }),
+        });
+      }
+      setLoading(false);
```

Comment:
nit: could be slightly more defensive and put `setLoading` inside a `finally` block for the case that the `toasts.addError` also throws

Reply:
I sometimes even forget `finally` is a thing in javascript.